storage/log_storage.go

```diff
@@ -55,12 +70,25 @@ type ReadOnlyLogTreeTX interface {
 // released any resources owned by the LogTX.
 // A LogTreeTX can only modify the tree specified in its creation.
 type LogTreeTX interface {
-	TreeTX
-	LogRootReader
-	LogRootWriter
-	LeafReader
-	LeafQueuer
-	LeafDequeuer
+	ReadOnlyLogTreeTX
+	TreeWriter
+
+	// StoreSignedLogRoot stores a freshly created SignedLogRoot.
+	StoreSignedLogRoot(ctx context.Context, root trillian.SignedLogRoot) error
+	// QueueLeaves enqueues leaves for later integration into the tree.
+	// If error is nil, the returned slice of leaves will be the same size as the
+	// input, and each entry will hold:
+	//  - the existing leaf entry if a duplicate has been submitted
+	//  - nil otherwise.
+	// Duplicates are only reported if the underlying tree does not permit duplicates, and are
+	// considered duplicate if their leaf.LeafIdentityHash matches.
+	QueueLeaves(ctx context.Context, leaves []*trillian.LogLeaf, queueTimestamp time.Time) ([]*trillian.LogLeaf, error)
+	// DequeueLeaves will return between [0, limit] leaves from the queue.
+	// Leaves which have been dequeued within a Rolled-back Tx will become available for dequeing again.
+	// Leaves queued more recently than the cutoff time will not be returned. This allows for
+	// guard intervals to be configured.
+	DequeueLeaves(ctx context.Context, limit int, cutoffTime time.Time) ([]*trillian.LogLeaf, error)
+	UpdateSequencedLeaves(ctx context.Context, leaves []*trillian.LogLeaf) error
```

Comment:
Can you also add a comment to this method? Something similar to the one in #975.

Reply:
Done.